node/src/blockchain/errors/validation_status.rs

```diff
@@ -151,4 +180,67 @@ mod tests {
             .get(&BlockchainErrorKind::AppRpc(AppRpcErrorKind::Signing));
         assert_eq!(other_error_stats, None);
     }
+
+    #[test]
+    fn previous_attempts_hash_works_correctly() {
+        let now = SystemTime::now();
+        let clock = ValidationFailureClockMock::default()
+            .now_result(now)
+            .now_result(now)
+            .now_result(now + Duration::from_secs(2));
+        let attempts1 = PreviousAttempts::new(
+            BlockchainErrorKind::AppRpc(AppRpcErrorKind::Decoder),
+            &clock,
+        );
+        let attempts2 = PreviousAttempts::new(
+            BlockchainErrorKind::AppRpc(AppRpcErrorKind::Decoder),
+            &clock,
+        );
+        let attempts3 =
+            PreviousAttempts::new(BlockchainErrorKind::AppRpc(AppRpcErrorKind::Io), &clock);
+        let hash1 = {
+            let mut hasher = DefaultHasher::new();
+            attempts1.hash(&mut hasher);
+            hasher.finish()
+        };
+        let hash2 = {
+            let mut hasher = DefaultHasher::new();
+            attempts2.hash(&mut hasher);
+            hasher.finish()
+        };
+        let hash3 = {
+            let mut hasher = DefaultHasher::new();
+            attempts3.hash(&mut hasher);
+            hasher.finish()
+        };
+
+        assert_eq!(hash1, hash2);
+        assert_ne!(hash1, hash3);
```

Comment:
You should include a fourth which should be asserted as `hash4 != hash3 && hash4 != hash1`